inst/include/distributions/functors/lognormal_lpdf.hpp

```diff
@@ -90,12 +85,20 @@ namespace fims_distributions
                 FIMS_SIMULATE_F(this->of)
                 { // preprocessor definition in interface.hpp
                     // this simulates data that is mean biased
-                    if(this->input_type == "data"){
-                      this->observed_values->at(i) = fims_math::exp(rnorm(this->expected_values.get_force_scalar(i),
-                                                                          fims_math::exp(log_sd.get_force_scalar(i))));
-                    } else {
-                      this->x[i] = fims_math::exp(rnorm(this->expected_values.get_force_scalar(i),
-                                                        fims_math::exp(log_sd.get_force_scalar(i))));
+                  if(this->input_type == "data"){                      
+                    this->observed_values->at(i) = 
+                      fims_math::exp(rnorm(this->get_expected(i),
+                      fims_math::exp(log_sd.get_force_scalar(i))));
+                  }
+                  if(this->input_type == "random_effects"){
+                    (*this->re)[i] = 
+                      fims_math::exp(rnorm(this->get_expected(i),
+                      fims_math::exp(log_sd.get_force_scalar(i))));
+                  }
+                  if(this->input_type == "prior"){
+                    (*(this->priors[i]))[0] = 
+                      fims_math::exp(rnorm(this->get_expected(i),
+                      fims_math::exp(log_sd.get_force_scalar(i))));
```

Comment:
The code coverage indicates that we do not have tests for priors, is that true?

Reply:
Current tests set up normal priors, not lognormal. The C++ test [here](https://github.com/NOAA-FIMS/FIMS/blob/dev-random-effects-rebase/tests/gtest/test_info_setup_priors.cpp) can be expanded to include a lognormal prior.

Reply:
@Andrea-Havron-NOAA can you add a test for the lognormal as you indicated above and also increase the comments in the testing files to indicate what they are doing? After that, we can squash all of the commits into one again and merge this into dev.

Reply:
[this](https://github.com/NOAA-FIMS/FIMS/blob/dev-random-effects-rebase/inst/include/distributions/functors/density_components_base.hpp#L56) method should return Type& rather than Type if a reference is the desired return type, otherwise it's returning a copy.